fullrt/dht.go

```diff
@@ -834,35 +834,38 @@ func (dht *FullRT) execOnMany(ctx context.Context, fn func(context.Context, peer
 	putctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	waitAllCh := make(chan struct{}, len(peers))
+	waitFailCh := make(chan struct{})
+	waitSuccessCh := make(chan struct{})
 	numSuccessfulToWaitFor := int(float64(len(peers)) * dht.waitFrac)
-	waitSuccessCh := make(chan struct{}, numSuccessfulToWaitFor)
 	for _, p := range peers {
 		go func(p peer.ID) {
 			fnCtx, fnCancel := context.WithTimeout(putctx, dht.timeoutPerOp)
 			defer fnCancel()
 			err := fn(fnCtx, p)
 			if err != nil {
 				logger.Debug(err)
+				waitFailCh <- struct{}{}
 			} else {
 				waitSuccessCh <- struct{}{}
 			}
-			waitAllCh <- struct{}{}
 		}(p)
 	}
 
-	numSuccess, numDone := 0, 0
+	var numFail, numSuccess int
 	t := time.NewTimer(time.Hour)
```

Comment:
in reality this shouldn't be a problem, but we should probably stop this goroutine (and empty it). Then we can do a reset if numSuccess == numSuccessfulToWaitFor (i.e. the first time it's hit), and then on subsequent (>=) successes we can do the stop + empty before the reset.

Sound reasonable?

Reply:
So, you are saying that the initial wait for enough successes should be forever, instead of the 1 hour it is currently set to -- which means all `dht.timeoutPerOp` is the only timeout?

I think that sounds OK, and probably does not matter either way in reality.  I assumed the one-hour initial timer timeout was a safety measure in case `dht.timeoutPerOp` was set to something unreasonably long.

Reply:
Yep, it shouldn't matter anyway but less magic constants to think about is probably better. If the user is ok with their query taking forever and not timing out that's on them. An hour is kind of a nonsense number (I chose it at the time because it was the largest built in duration constant).

Reply:
Done